### PR TITLE
Add setting to enable/disable new user registration

### PR DESCRIPTION
**Pull request by InspectorCaracal**
#### Brief overview of PR changes/additions
Adds a single new setting, `REGISTER_ENABLED` which enables registration. By default it's set to `True` but if set to `False`, it deactivates the new account command and the registration page on the website.

#### Motivation for adding to Evennia
Adding a useful requested feature

### CODE REVIEW

File: evennia/settings_default.py [hunk 1]
```diff
@@ -34,6 +34,8 @@
 # Lockdown mode will cut off the game from any external connections
 # and only allow connections from localhost. Requires a cold reboot.
 LOCKDOWN_MODE = False
+# Enables new account registration
```
Comment (Griatch) [inline evennia/settings_default.py:37]: I would suggest a clearer comment here, something like
```python
# Control new registrations with the default unloggedin-create command. 
# Set to False to lock down your game to newcomers.
```

File: evennia/settings_default.py [hunk 2]
```diff
@@ -34,6 +34,8 @@
...
+REGISTER_ENABLED = True
```
Comment (Griatch) [inline evennia/settings_default.py:38]: This settings name is a bit ambivalent. I suggest being more explicit, like `NEW_ACCOUNT_REGISTRATION_ENABLED`

**Review by Griatch — CHANGES_REQUESTED** [review]
This is a nice feature! Looks like a clean implementation. I think it would be good with a unit test (using `@override_settings` decorator) to showcase this working for `CmdUnconnectedCreate` command though.

**Comment by InspectorCaracal** [comment]
> This is a nice feature! Looks like a clean implementation. I think it would be good with a unit test (using `@override_settings` decorator) to showcase this working for `CmdUnconnectedCreate` command though.

Oh, good catch! I'll add that today along with the setting name and comments changes.

File: evennia/settings_default.py (hunk 1 above)
Comment (InspectorCaracal) [reply evennia/settings_default.py]: I went with:
```py
# Controls whether new account registration is available.
# Set to False to lock down the registration page and the create account command. 
```

**Review by Griatch — APPROVED** [review]
Thanks for the changes!

**Pull request merged**